pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -134,6 +136,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 		return nil, fmt.Errorf("cannot parse node port: %v", err)
 	}
 
+	pluginName := options.NetworkConfig.NetworkPluginName
+	if sdnovs.IsOpenShiftNetworkPlugin(pluginName) {
+		// Detect any plugin mismatches between node and master
+		clusterNetwork, err := originClient.ClusterNetwork().Get(sdnapi.ClusterNetworkDefault)
+		if kerrs.IsNotFound(err) {
+			return nil, fmt.Errorf("detected network plugin mismatch between OpenShift node(%q) and master(non OpenShift network plugin)", pluginName)
```

Comment:
Message would be clearer if it said: `master has not created a default cluster network, network plugin %q can not start`